impls/hack/bin/step0_repl.hack

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env hhvm -c .hhvmconfig.hdf
+
+namespace Mal\Step0;
+
+require_once(__DIR__.'/../vendor/autoload.hack');
+
+<<__EntryPoint>>
+async function main_async(): Awaitable<void> {
+  AutoloadMap\initialize();
+
+  $cli_input = IO\request_input();
+  while (true) {
+    // Handles CTRL+D
+    if ($cli_input->isEndOfFile()) {
+      return;
+    }
+    echo "user> ";
+    // HHAST_IGNORE_ERROR[DontAwaitInALoop]
+    echo rep(await $cli_input->readLineAsync());
+  }
+}
+
+
+function rep(string $input): string {
+  return print_out(evaluate(read($input)));
+}
+
+// READ
+function read(string $input): string {
```

Comment:
I didn't capitalize these before the Hack linter would complain about the names.

Reply:
That's fine, the reason these are typed in allcaps is because `read`, `eval` and `print` tend to be already used function names in programming languages and avoids the conflict between the built-in name and our own.